src/query/range_query/fast_field_range_query.rs

```diff
@@ -0,0 +1,212 @@
+use std::ops::RangeInclusive;
+use std::sync::Arc;
+
+use fastfield_codecs::Column;
+
+use crate::fastfield::{MakeZero, MultiValuedFastFieldReader};
+use crate::{DocId, DocSet, TERMINATED};
+
+/// Helper to have a cursor over a vec of docids
+struct VecCursor {
+    docs: Vec<u32>,
+    current_pos: usize,
+}
+impl VecCursor {
+    fn new() -> Self {
+        Self {
+            docs: Vec::with_capacity(32),
+            current_pos: 0,
+        }
+    }
+    fn next(&mut self) -> Option<u32> {
+        self.current_pos += 1;
+        self.current()
+    }
+    #[inline]
+    fn current(&self) -> Option<u32> {
+        self.docs.get(self.current_pos).copied()
+    }
+    fn get_cleared_data(&mut self) -> &mut Vec<u32> {
```

Comment:
That's a very strange API.
I'm not sure where you are going with this.

Could we use `Into<Vec<u32>>` instead?